### PR TITLE
Implement Use command with priority removal

### PR DESCRIPTION
**Pull request by yakultbottle**
Priority removal in this case refers to always removing items with an earlier expiry date first

### CODE REVIEW

File: src/main/java/seedu/pill/util/ItemMap.java [hunk 1]
```diff
@@ -399,6 +399,140 @@ public ItemMap getExpiringItems(LocalDate cutOffDate) {
...
+     * @param quantityToUse the quantity of the item to consume
+     * @throws PillException if:
+     *         - the requested quantity exceeds the total available stock for the item
+     *         - the specified item name does not exist in the inventory
+     */
+    public void useItem(String itemName, int quantityToUse) throws PillException {
+        if (quantityToUse > this.stockCount(itemName)) {
+            LOGGER.warning("Attempt to use more items than available: name=" + itemName +
+                    ", requested=" + quantityToUse +
+                    ", available=" + this.stockCount(itemName));
```
Comment (cxc0418) [inline src/main/java/seedu/pill/util/ItemMap.java:417]: I like how detailed the logging is here, especially in tracking when items are completely or partially used

File: src/main/java/seedu/pill/util/Parser.java [hunk 2]
```diff
@@ -278,6 +282,46 @@ && isValidDate(splitArguments[quantityIndex + 1])) {
...
+    private UseItemCommand parseUseItemCommand(String arguments) throws PillException {
+        String[] splitArguments = arguments.split("\\s+");
+
+        if (splitArguments.length < 2) {
+            throw new PillException(ExceptionMessages.INVALID_USE_COMMAND);
+        }
+
+        StringBuilder itemNameBuilder = new StringBuilder();
+        int currentIndex = 0;
+        int quantity = 1; // default use amount is 1
```
Comment (cxc0418) [inline src/main/java/seedu/pill/util/Parser.java:306]: I like how a default quantity is set to 1 if not specified in the input.

File: src/test/java/seedu/pill/command/UseItemCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/**
+ * Unit tests for UseItemCommand class.
+ */
+public class UseItemCommandTest {
+    private ItemMap itemMap;
+    private Storage storage;
```
Comment (cxc0418) [inline src/test/java/seedu/pill/command/UseItemCommandTest.java:21]: I like how each test case is named clearly based on its purpose. Good use of assertions for validating item quantities after each command execution. Thank you for your effort.

**Review by cxc0418 — APPROVED** [review]

**Comment by yakultbottle** [comment]
Closes #82

**Pull request merged**